### PR TITLE
Remove IHostingEnvironment dependency

### PR DESCRIPTION
**Pull request by Tratcher**
https://github.com/aspnet/Security/issues/1201#issuecomment-299541453
This functionality is moving into Hosting. Expect a Hosting PR shortly.

### CODE REVIEW

File: src/Microsoft.AspNetCore.DataProtection/DataProtectionUtilityExtensions.cs [hunk 1]
```diff
@@ -37,10 +36,6 @@ public static string GetApplicationUniqueIdentifier(this IServiceProvider servic
             if (services != null)
             {
                 discriminator = services.GetService<IApplicationDiscriminator>()?.Discriminator;
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.DataProtection/DataProtectionUtilityExtensions.cs]: Make this GetRequiredService

Comment (JunTaoLuo) [inline src/Microsoft.AspNetCore.DataProtection/DataProtectionUtilityExtensions.cs]: Hmm... Breaks a few tests. I'll merge this now and maybe fix the tests later.

Comment (PinpointTownes) [inline src/Microsoft.AspNetCore.DataProtection/DataProtectionUtilityExtensions.cs]: @davidfowl not really a good idea. Even if this service is now registered by the hosting block, there are cases where no `IApplicationDiscriminator` will be available: when using DP in a non-ASP.NET Core app or in an OWIN/Katana app using the interop package.

**Comment by PinpointTownes** [comment]
Woot, nice. This even comes with a bonus: OWIN/Katana apps using `Microsoft.Owin.Security.Interop` will no longer bring the ASP.NET Core HTTP/Hosting abstractions that were not really needed :+1:

File: src/Microsoft.AspNetCore.DataProtection/DataProtectionUtilityExtensions.cs (hunk 1 above)
Comment (davidfowl) [inline src/Microsoft.AspNetCore.DataProtection/DataProtectionUtilityExtensions.cs]: @PinpointTownes as long as it works with null (which it seems like it does) I guess that's fine.

Comment (PinpointTownes) [inline src/Microsoft.AspNetCore.DataProtection/DataProtectionUtilityExtensions.cs]: This service won't be automatically registered when using DP without the hosting stack so `GetRequiredService` will throw an exception an prevent the app from loading/working correctly.

**Comment by Eilon** [comment]
Yup, that was always a weird dependency.

**Pull request merged**